cmd/antrea-agent/options.go

```diff
@@ -167,12 +215,68 @@ func (o *Options) setDefaults() {
 		o.config.APIPort = apis.AntreaAgentAPIPort
 	}
 	if o.config.NodeType == "" {
-		o.config.NodeType = defaultNodeType.String()
+		o.config.NodeType = config.K8sNode.String()
 	}
-	if o.config.NodeType == config.K8sNode.String() {
-		o.setK8sNodeDefaultOptions()
-	} else {
-		o.setExternalNodeDefaultOptions()
+	if o.config.CNISocket == "" {
+		o.config.CNISocket = cni.AntreaCNISocketAddr
+	}
+	if o.config.HostGateway == "" {
+		o.config.HostGateway = defaultHostGateway
+	}
+	if o.config.TrafficEncapMode == "" {
+		o.config.TrafficEncapMode = config.TrafficEncapModeEncap.String()
```

Comment:
I saw that function `setExternalNodeDefaultOptions` is removed, and there is no change to update `o.config.TrafficEncapMode` in the code, it may lead to encap mode is used in ExternalNode scenario with the current configurations in file [ default configurati](https://github.com/antrea-io/antrea/blob/main/build/yamls/externalnode/conf/antrea-agent.conf). 

ExternalNode does not support encap mode.